lib/src/components/dropdown/dropdown.dart

```diff
@@ -0,0 +1,440 @@
+    this.checkBoxType,
+  });
+
+  /// {@macro zeta-component-rounded}
+  final bool rounded;
+
+  /// If [ZetaDropdownItem] is selected
+  final bool selected;
+
+  /// Value of [ZetaDropdownItem]
+  final String value;
+
+  /// Leading icon for [ZetaDropdownItem]
+  final Icon? leadingIcon;
+
+  /// Handles clicking for [ZetaDropdownItem]
+  final VoidCallback? onPress;
+
+  /// If checkbox is to be shown, the type of it.
+  final CheckBoxType? checkBoxType;
+
+  /// Returns copy of [ZetaDropdownItem] with those private variables included
+  ZetaDropdownItem copyWith({
+    bool? round,
+    bool? focus,
+    CheckBoxType? boxType,
+    VoidCallback? press,
+    GlobalKey? itemKey,
+  }) {
+    return ZetaDropdownItem._(
+      rounded: round ?? rounded,
+      selected: focus ?? selected,
+      onPress: press ?? onPress,
+      checkBoxType: boxType ?? checkBoxType,
+      value: value,
+      leadingIcon: leadingIcon,
+      key: itemKey ?? key,
+    );
+  }
+
+  @override
+  State<ZetaDropdownItem> createState() => _ZetaDropdownMenuItemState();
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty<bool>('rounded', rounded))
+      ..add(DiagnosticsProperty<bool>('selected', selected))
+      ..add(StringProperty('value', value))
+      ..add(ObjectFlagProperty<VoidCallback?>.has('onPress', onPress))
+      ..add(EnumProperty<CheckBoxType?>('checkBoxType', checkBoxType));
+  }
+}
+
+class _ZetaDropdownMenuItemState extends State<ZetaDropdownItem> {
+  final controller = MaterialStatesController();
+
+  @override
+  void initState() {
+    super.initState();
+    controller.addListener(() {
+      if (context.mounted &&
+          mounted &&
+          !controller.value.contains(MaterialState.disabled)) {
+        setState(() {});
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final colors = Zeta.of(context).colors;
+
+    return DefaultTextStyle(
+      style: ZetaTextStyles.bodyMedium,
+      child: OutlinedButton(
+        key: widget.key,
+        onPressed: widget.onPress,
+        style: _getStyle(colors),
+        child: Row(
+          children: [
+            const SizedBox(width: ZetaSpacing.x3),
+            if (widget.checkBoxType != CheckBoxType.none)
+              Checkbox(
+                value: widget.selected,
+                shape: RoundedRectangleBorder(
+                  borderRadius: widget.checkBoxType == CheckBoxType.square
+                      ? ZetaRadius.none
+                      : ZetaRadius.full,
+                ),
+                onChanged: (val) {
+                  widget.onPress!.call();
+                },
+              )
+            else
+              widget.leadingIcon ??
+                  const SizedBox(
+                    width: 24,
+                  ),
+            const SizedBox(width: ZetaSpacing.x3),
+            Text(
```

Comment:
I think you need to check the text styling here against the figma as well as checking the spacing between the items in the menu